lib/spack/docs/build_systems.rst

```diff
@@ -0,0 +1,78 @@
+
+.. _build-systems:
+
+=============
+Build Systems
+=============
+
+This guide provides information specific to a particular
+`build system <https://en.wikipedia.org/wiki/List_of_build_automation_software>`_.
+It assumes knowledge of general Spack packaging capabilities and expands
+on these ideas for each distinct build system that Spack supports:
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Make-based tools
+
+   build_systems/makefilepackage
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Build-script generation tools
+
+   build_systems/autotoolspackage
+   build_systems/cmakepackage
+   build_systems/qmakepackage
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Language-specific tools
+
+   build_systems/pythonpackage
+   build_systems/rpackage
+   build_systems/perlpackage
+   build_systems/rubypackage
+   build_systems/octavepackage
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Non-Make-based tools
+
+   build_systems/sconspackage
+   build_systems/wafpackage
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Other
+
+   build_systems/intelpackage
+   build_systems/cudapackage
+   build_systems/custompackage
+
+For reference, the :py:mod:`Build System API docs <spack.build_systems>`
+provide a list of build systems and methods/attributes that can be
+overridden. If you are curious about the implementation of a particular
+build system, you can view the source code by running:
+
+.. code-block:: console
+
+   $ spack edit --build-system autotools
```

Comment:
Let's not reopen old wounds. This was settled on in #2865. You might not find it useful, but I probably used it hundreds of times while writing these docs. I don't know if I could survive without it.

Reply:
Please just provide appropriate alternatives.  Tell people where the file is located in case they want to edit it themselves, give URLs to GItHub, etc.